lib/messages/blob.js

```diff
@@ -19,32 +20,51 @@ class Blob extends Message {
   }
 
   getCore() {
-    let lines = Buffer.from(this.blob.content, 'base64').toString().split(/\n/);
+    const buffer = Buffer.from(this.blob.content, 'base64');
     let title = this.blob.path;
     let htmlURL = this.blob.html_url;
 
-    if (this.start) {
-      lines = lines.slice(
-        Number(this.start) - 1,
-        Number(this.end || this.start),
-      );
+    // only unfurl binary files that are images
+    if (isBinaryFile.sync(buffer, this.blob.size)) {
+      const imageExtensions = ['png', 'jpg', 'jpeg', 'gif'];
+      const blobExtension = this.blob.name.toLowerCase().split('.')[1];
 
-      title += `:${this.start}`;
-      htmlURL += `#L${this.start}`;
+      if (imageExtensions.includes(blobExtension)) {
+        return {
+          fallback: `[${this.repository.full_name}] ${title}`,
+          title,
+          title_link: htmlURL,
+          image_url: this.blob.download_url,
+        };
+      } else { // eslint-disable-line no-else-return
+        throw new Error('File is binary');
+      }
```

Comment:
I opted for disabling the `no-else-return` here because it made the implementation look weird and not as readable when the `else` block was omitted (in that block of code). 

```js
if (imageExtensions.includes(blobExtension)) {
	return {
       fallback: `[${this.repository.full_name}] ${title}`,
       title,
       title_link: htmlURL,
       image_url: this.blob.download_url,
     };
}
throw new Error('File is binary');
```
Seeing that we are only throwing the error _for now_, I see this as temporary, but I can easily remove it and follow the rule.